### PR TITLE
Boats: Check player attached object

### PR DESCRIPTION
**Pull request by Rui914**
#643

### CODE REVIEW

**Comment by kilbith** [comment]
A bit of attention to this PR please.

**Comment by tenplus1** [comment]
Thanks Rui914, I've added this to Xanadu's boat mod and works fine :)

**Comment by paramat** [comment]
@tenplus1 please can you explain what it does? so i can test and +1 it.

File: mods/boats/init.lua [hunk 1]
```diff
@@ -62,6 +69,10 @@ function boat.on_rightclick(self, clicker)
 			clicker:setpos(pos)
 		end)
 	elseif not self.driver then
+		if handlers[name] and handlers[name].driver then
+			handlers[name].driver = nil
+		end
+		handlers[name] = self.object:get_luaentity()
```
Comment (ShadowNinja) [inline mods/boats/init.lua]: `self.object:get_luaentity()` == `self`

**Comment by tenplus1** [comment]
paramat: at the moment a player can place 2 boats in water, right-click on one then the other and take control of both boats, the change adds a small handler to the boats so that a player can only control one at a time.

**Comment by paramat** [comment]
Thanks.

**Comment by Ferk** [comment]
Perhaps it would be best to have an API call that indicates whether a user is attached to something. 
Then we don't need to keep handlers every time we want to keep track if a user is already attached somewhere.

**Comment by Ferk** [comment]
Oh wait.. there IS a get_attach: https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L2512

Why is this not used? :S

**Comment by Rui914** [comment]
Update.
(Thanks @Ferk)

**Comment by Ferk** [comment]
Looks nice!

IMHO, even more versatile if you added and called a "luaentity.is_player_attached(player)" instead of simply "luaentity.driver"

This way it would be possible to cover vehicles that have more than 1 player attached, if they also implemented an is_player_attached function (and imho, if such function doesn't exist I would rather default having the boat do nothing when the attached entity was found than going ahead attaching the player, what if there's a vehicle that doesn't play nice and names his "driver" something else?).

But this is already nitpicking.

**Comment by paramat** [comment]
:+1:  from PilzAdam and sfan5 http://irc.minetest.ru/minetest-dev/2015-10-29#i_4435553
Now tested.

**Comment by paramat** [comment]
a88576afd1a27faf1a8036e2ed4c4b1e5c78ec90

**Pull request closed without merging**